### PR TITLE
Make Accept-Encoding header handling thread-safe

### PR DESCRIPTION
**Pull request by coxlinton**

### CODE REVIEW

File: httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/ContentCompressionExec.java [hunk 1]
```diff
@@ -70,16 +71,20 @@
...
     private final Lookup<InputStreamFactory> decoderRegistry;
     private final boolean ignoreUnknown;
 
     public ContentCompressionExec(
             final List<String> acceptEncoding,
             final Lookup<InputStreamFactory> decoderRegistry,
             final boolean ignoreUnknown) {
-        this.acceptEncoding = acceptEncoding != null ? acceptEncoding.toArray(
-                new String[acceptEncoding.size()]) : new String[] {"gzip", "x-gzip", "deflate"};
+        // MessageSupport.format is not thread-safe, because it sorts the
```
Comment (ok2c) [inline httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/ContentCompressionExec.java]: @coxlinton I will fix `MessageSupport#format` in HttpCore master. Do you mind dropping the comment about `MessageSupport#format` thread safety from this change-set?

Comment (coxlinton) [reply httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/ContentCompressionExec.java]: Not at all - that definitely seems like an excellent idea. Updated made.

Comment (ok2c) [reply httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/ContentCompressionExec.java]: @coxlinton Could you please squash those two commits into a single one?

Comment (coxlinton) [reply httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/ContentCompressionExec.java]: Sorry about that - done.

**Pull request merged**